### PR TITLE
[Feature] Support different filename templates in GenerateSegmentIndices

### PR DESCRIPTION
**Pull request by ckkelvinchan**
## Motivation
As mentioned in issue #323, there will be errors if the file names are not formated as `{v:08d}.png`. This may cause problems if one use custom filename format.

## Modification
In this PR, we modify `GenerateSegmentIndices` so that one can use different filename formats. But note that the frame indices must start from 0.

### CODE REVIEW

File: mmedit/apis/restoration_video_inference.py [hunk 1]
```diff
@@ -18,24 +18,25 @@ def pad_sequence(data, window_size):
     return data
 
 
-def restoration_video_inference(model, img_dir, window_size):
+def restoration_video_inference(model, img_dir, window_size, filename_tmpl):
```
Comment (innerlee) [inline mmedit/apis/restoration_video_inference.py:21]: Have you changed all references to this function?

Comment (innerlee) [reply mmedit/apis/restoration_video_inference.py:21]: Okay it seems so

File: mmedit/apis/restoration_video_inference.py [hunk 2]
```diff
@@ -18,24 +18,25 @@ def pad_sequence(data, window_size):
...
     """Inference image with the model.
-
```
Comment (innerlee) [inline mmedit/apis/restoration_video_inference.py:23]: Do not remove the space

File: mmedit/apis/restoration_video_inference.py (hunk 1 above)
Comment (ckkelvinchan) [reply mmedit/apis/restoration_video_inference.py:21]: Yes, only `restoration_video_demo.py` uses this fuction.

File: mmedit/apis/restoration_video_inference.py [hunk 3]
```diff
@@ -27,15 +27,18 @@ def restoration_video_inference(model, img_dir, window_size):
         window_size (int): The window size used in sliding-window framework.
             This value should be set according to the settings of the network.
             A value smaller than 0 means using recurrent framework.
-
```
Comment (innerlee) [inline mmedit/apis/restoration_video_inference.py:30]: and here, of course

File: mmedit/datasets/pipelines/augmentation.py [hunk 4]
```diff
@@ -964,11 +966,13 @@ def __call__(self, results):
         lq_path_root = results['lq_path']
         gt_path_root = results['gt_path']
         lq_path = [
-            osp.join(lq_path_root, clip_name, f'{v:08d}.png')
+            osp.join(lq_path_root, clip_name,
+                     f'{self.filename_tmpl.format(v)}.png')
```
Comment (innerlee) [inline mmedit/datasets/pipelines/augmentation.py:970]: Question: should `.png` be part of the template?

Comment (ckkelvinchan) [reply mmedit/datasets/pipelines/augmentation.py:970]: We usually use `.png`, but I think that could work if someone wants to use `jpg` as inputs. I will modify that.

File: mmedit/datasets/pipelines/augmentation.py [hunk 5]
```diff
@@ -964,12 +966,12 @@ def __call__(self, results):
         lq_path_root = results['lq_path']
         gt_path_root = results['gt_path']
         lq_path = [
-            osp.join(lq_path_root, clip_name, f'{v:08d}.png')
-            for v in neighbor_list
+            osp.join(lq_path_root, clip_name,
+                     f'{self.filename_tmpl.format(v)}') for v in neighbor_list
```
Comment (innerlee) [inline mmedit/datasets/pipelines/augmentation.py:970]: In this case, the f-str is redundant

Comment (ckkelvinchan) [reply mmedit/datasets/pipelines/augmentation.py:970]: Oh you are right. I missed that.

**Review by innerlee — APPROVED** [review]

**Pull request merged**